File: leetcode/53. Maximum Subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int max_sub = INT_MIN;
        int curr = 0;
        for(auto num: nums){
            curr += num;
            max_sub = max(max_sub, curr);
            if(curr <= 0){
                curr = 0;
            }
        }
        return max_sub;
    }
};

File: leetcode/53. Maximum Subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        curr = 0
        min_val = 0
        res = nums[0]
        for num in nums:
            curr += num
            res = max(res, curr-min_val)
            min_val = min(min_val, curr)
        return res
